anta/reporter/__init__.py

```diff
@@ -102,35 +103,84 @@ def _color_result(self, status: AntaTestStatus) -> str:
         color = RICH_COLOR_THEME.get(str(status), "")
         return f"[{color}]{status}" if color != "" else str(status)
 
-    def report_all(self, manager: ResultManager, title: str = "All tests results") -> Table:
-        """Create a table report with all tests for one or all devices.
+    TITLE_ALL = "All tests results"
 
-        Create table with full output: Device | Test Name | Test Status | Message(s) | Test description | Test category
+    def report_expanded(self, manager: ResultManager) -> Table:
+        """Create a table report with all tests.
+
+        Create table with columns: Category | Test | Device | Description | Status | Message(s) | Inputs
 
         Parameters
         ----------
         manager
             A ResultManager instance.
-        title
-            Title for the report. Defaults to 'All tests results'.
 
         Returns
         -------
         Table
             A fully populated rich `Table`.
         """
-        table = Table(title=title, show_lines=True)
-        headers = ["Device", "Test Name", "Test Status", "Message(s)", "Test description", "Test category"]
+        table = Table(title=ReportTable.TITLE_ALL, show_lines=True)
+        headers = ["Category", "Test", "Device", "Description", "Status", "Message(s)", "Inputs"]
         table = self._build_headers(headers=headers, table=table)
 
-        def add_line(result: TestResult) -> None:
+        def add_line(result: TestResult | AtomicTestResult, name: str | None = None) -> None:
+            categories = device = test = None
+            if isinstance(result, TestResult):
+                categories = ", ".join(convert_categories(result.categories))
+                device = str(result.name)
+                test = result.test
+            else:
+                test = name
+            state = self._color_result(result.result)
+            message = self._split_list_to_txt_list(result.messages) if len(result.messages) > 0 else ""
+            inputs = result.inputs.model_dump_json(indent=2, exclude={"result_overwrite", "filters"}) if result.inputs is not None else None
```

Comment:
I feel like like having all inputs at the parent test level is unnesserary when we have atomic results. We are just repeating them. Maybe we should add them only if atomic results are missing.

Reply:
Actually it depends on the test implementation. One can decide to have atomic results but no inputs tied to these atomic results.
Also, the parent test can have more input than the union of all the atomic results inputs.
But in most test implementation that will be redundant. Let's just remove the parent test inputs.